backend/service/service_gov_params.go

```diff
@@ -28,188 +29,227 @@ func (service *GovParamsService) QueryAll() []document.GovParams {
+		switch curV := currentMap[k].(type) {
+		case string:
+			currentValueStr = curV
+		}
+
+		tmp := document.GovParams{
+			Module:       lcd.GovModuleDistr,
+			Key:          k,
+			Range:        v.Range,
+			Description:  v.Description,
+			GenesisValue: genValueStr,
+			CurrentValue: currentValueStr,
+		}
+		result = append(result, tmp)
+	}
+	return result, nil
+}
+
+func (_ GovParamsService) GetGovMintParamList(genesisMap, currentMap map[string]interface{}) ([]document.GovParams, error) {
+	mintKeyRangeMap := lcd.GetMintKeyWithRangeMap()
+	result := make([]document.GovParams, 0, len(mintKeyRangeMap))
+	for k, v := range mintKeyRangeMap {
+		genValueStr := ""
+		switch genV := genesisMap[k].(type) {
+		case string:
+			genValueStr = genV
+		}
+		currentValueStr := ""
+		switch curV := currentMap[k].(type) {
+		case string:
+			currentValueStr = curV
+		}
+
+		tmp := document.GovParams{
+			Module:       lcd.GovModuleMint,
+			Key:          k,
+			Range:        v.Range,
+			Description:  v.Description,
+			GenesisValue: genValueStr,
+			CurrentValue: currentValueStr,
+		}
+		result = append(result, tmp)
+	}
+	return result, nil
+}
+
+func (_ GovParamsService) GetGovSlashingParamList(genesisMap, currentMap map[string]interface{}) ([]document.GovParams, error) {
+	slashingKeyRangeMap := lcd.GetSlashingKeyWithRangeMap()
+	result := make([]document.GovParams, 0, len(slashingKeyRangeMap))
+	for k, v := range slashingKeyRangeMap {
+		genValueStr := ""
+		switch genV := genesisMap[k].(type) {
+		case string:
+			genValueStr = genV
+		}
+		currentValueStr := ""
+		switch curV := currentMap[k].(type) {
+		case string:
+			currentValueStr = curV
+		}
+
+		tmp := document.GovParams{
+			Module:       lcd.GovModuleSlashing,
+			Key:          k,
+			Range:        v.Range,
+			Description:  v.Description,
+			GenesisValue: genValueStr,
+			CurrentValue: currentValueStr,
+		}
+		result = append(result, tmp)
+	}
+	return result, nil
+}
+
+func (gov GovParamsService) GetDbInitGovModuleParamList(genesisMap, currentMap map[string]interface{}) ([]document.GovParams, error) {
+
+	authList, err := gov.GetGovAuthParamList(genesisMap, currentMap)
+	if err != nil {
+		return nil, err
+	}
+
+	stakeList, err := gov.GetGovStakeParamList(genesisMap, currentMap)
+	if err != nil {
+		return nil, err
+	}
+
+	stakeList = append(stakeList, authList...)
+
+	distrList, err := gov.GetGovDistrParamList(genesisMap, currentMap)
+	if err != nil {
+		return nil, err
+	}
+
+	distrList = append(distrList, stakeList...)
+
+	mintList, err := gov.GetGovMintParamList(genesisMap, currentMap)
+	if err != nil {
+		return nil, err
+	}
+
+	mintList = append(mintList, distrList...)
+
+	slashingList, err := gov.GetGovSlashingParamList(genesisMap, currentMap)
+	if err != nil {
+		return nil, err
+	}
+	return append(slashingList, mintList...), nil
+}
+
+func (gov GovParamsService) UpdateCurrentValueByKey(kv map[string]interface{}) error {
+
+	bulk := getDb().C(document.CollectionNmGovParams).Bulk()
```

Comment:
did you close session after function return ?

Reply:
sorry, I forgot.